packages/greenproof-contracts/contracts/facets/IssuerFacet.sol

```diff
@@ -80,6 +85,33 @@ contract IssuerFacet is SolidStateERC1155, IGreenProof {
         LibIssuer.discloseData(dataHash, key, value);
     }
 
+    /**
+     * @notice approveOperator -  Grants approval to an operator to transfer certificates of a specific certificate owner
+     * @param operator The address of the operator being approved
+     * @param certificateOwner The address of the certificate owner for whom the operator is being approved
+     * @dev If the operator is already approved, the function will revert.
+     * @dev If the caller of this function does not have the `approver` role, the function will revert.
+     * @dev `msg.sender` cannot be the same as `operator`
+     */
+    function approveOperator(address operator, address certificateOwner) external onlyApprover {
+        LibIssuer.preventAlreadyApproved(operator, certificateOwner);
```

Comment:
doesn't it make more operations and cost more than just setting it every time?

Reply:
In general, reading a storage variable costs less than writing it. Since we will pay write cost anytime we set  the variable, it will be more expansive than just reading and reverting.
We indeed have more code by adding this check, but the overall runtime gas cost is better managed IMO.